Libraries/Text/RCTTextField.m

```diff
@@ -173,16 +192,19 @@ - (void)textFieldSubmitEditing
 
 - (void)textFieldBeginEditing
 {
-  if (_selectTextOnFocus) {
-    dispatch_async(dispatch_get_main_queue(), ^{
-      [self selectAll:nil];
-    });
-  }
   [_eventDispatcher sendTextEventWithType:RCTTextEventTypeFocus
                                  reactTag:self.reactTag
                                      text:self.text
                                       key:nil
                                eventCount:_nativeEventCount];
+
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Tested and it does what I mentioned above, on focus it sends an `onSelectionChange` event to JS with the text fully selected then JS can handle it like any other selection change event.